qa/L0_request_cancellation/scheduler_test.py

```diff
@@ -228,6 +243,15 @@ def test_scheduler_streaming_request_cancellation(self):
         time.sleep(2)  # ensure reaper thread has responded
         self._assert_streaming_response_is_cancelled(response)
 
+    def test_zerror_metrics(self):
```

Comment:
We should ideally have tests for all of the newly introduced failure `reasons`. So need tests for `rejected`, `backend`, and `other`. Feel free to lay out a rough test plan in text and I can review that too.